File: src/intro_tf2/src/intro_tf2_broadcaster.cpp
#include <ros/ros.h>
#include <tf2/LinearMath/Quaternion.h>
#include <tf2_ros/transform_broadcaster.h>
#include <geometry_msgs/TransformStamped.h>

class BroadcasterNode
{
public:
  BroadcasterNode()
    : tf2_br_() {}

  void broadcast_robot_pose(double x, double y, double theta)
  {
    geometry_msgs::TransformStamped transform_stamped;

    transform_stamped.header.stamp = ros::Time::now();
    transform_stamped.header.frame_id = "map";
    transform_stamped.child_frame_id = "base_link";
    transform_stamped.transform.translation.x = x;
    transform_stamped.transform.translation.y = y;
    transform_stamped.transform.translation.z = 0.0;
    tf2::Quaternion q;
    q.setRPY(0, 0, theta);
    transform_stamped.transform.rotation.x = q.x();
    transform_stamped.transform.rotation.y = q.y();
    transform_stamped.transform.rotation.z = q.z();
    transform_stamped.transform.rotation.w = q.w();

    tf2_br_.sendTransform(transform_stamped);
  }

  void spin()
  {
    ros::Rate rate(5);
    while (ros::ok())
    {
      broadcast_robot_pose(0,0,0);
      ros::spinOnce();
      rate.sleep();
    }
  }

private:
  tf2_ros::TransformBroadcaster tf2_br_;
};

int main(int argc, char** argv)
{
  ros::init(argc, argv, "tf2_broadcaster_node");
  BroadcasterNode node;
  node.spin();

  return EXIT_SUCCESS;
}

File: src/intro_tf2/src/intro_tf2_broadcaster.py
#!/usr/bin/env python
import rospy

# Because of transformations
import tf

import tf2_ros
import geometry_msgs.msg


class BroadcasterNode(object):
    def __init__(self):
        self._tf2_br = tf2_ros.TransformBroadcaster()

    def broadcast_robot_pose(self, x, y, theta):
        t = geometry_msgs.msg.TransformStamped()

        t.header.stamp = rospy.Time.now()
        t.header.frame_id = "map"
        t.child_frame_id = "base_link"
        t.transform.translation.x = x
        t.transform.translation.y = y
        t.transform.translation.z = 0.0
        q = tf.transformations.quaternion_from_euler(0, 0, theta)
        t.transform.rotation.x = q[0]
        t.transform.rotation.y = q[1]
        t.transform.rotation.z = q[2]
        t.transform.rotation.w = q[3]

        self._tf2_br.sendTransform(t)

    def spin(self):
        rate = rospy.Rate(5)
        while not rospy.is_shutdown():
            self.broadcast_robot_pose(0,0,0);
            rate.sleep()

if __name__ == '__main__':
    rospy.init_node('tf2_broadcaster_node')
    node = BroadcasterNode()
    node.spin()
